### ISSUE TITLE
eslint --fix behaves bizarrely with async in object literal where object method defined using old ES5 syntax

### ISSUE THREAD

**Issue by gitasong** [eslint/eslint#9679]
**Tell us about your environment**

* **ESLint Version:** 
From the `package.json`:
"eslint": "^2.7.0",
"eslint-config-airbnb": "^6.2.0",
"eslint-plugin-react": "^4.2.3",
 "eslint-plugin-react-native": "^1.0.0",
* **Node Version:** 8.9.1
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLInt

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser": "babel-eslint",
  "extends": "airbnb",
  "env": {
    "browser": false,
    "node": true
  },
  "globals": {
    "window": true,
    "fetch": true
  },
  "plugins": [
    "react",
    "react-native"
  ],
  "rules": {
    "comma-dangle": 0,
    "indent": [2, 2, {"SwitchCase": 1}],
    "no-console": 1,
    "react-native/no-unused-styles": 2,
    "no-return-assign": 0,
    "react/jsx-boolean-value": 0
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const obj = {
  objMethod: async function(arg1, arg2) {
  }
};
```

```bash
./node_modules/.bin/eslint --fix src/utilities/filename.js
```

**What did you expect to happen?**
That the original code would be preserved.

**What actually happened? Please include the actual, raw output from ESLint.**
Code output:
```
const obj = {
  *objMethodction(arg1, arg2) {
  }
};
```
`eslint --fix` a) adds an asterisk before the object method name, and b) deletes the following characters to collapse the method name and the keyword `function`: ` async fun` (note the leading space). This leads to a fatal parsing error in the original file, which I can't post for proprietary reasons:
```
/Users/home/Desktop/app/src/utilities/orig-filename.js
  154:19  error  Parsing error: await is a reserved word
```
I've validated that the following need to be present to trigger the error:
1) the object literal declaration (variable declaration followed by curly braces)
2) the object method name followed by a colon `:` and the keywords `async function` (i.e. the old ES5 syntax described in https://eslint.org/docs/rules/object-shorthand)

The error is not triggered by the new ES6 syntax for object methods, but there are no guidelines in the doc above on how to use that with `async`.

ESLint Errors in example (trivial, as these have to do with the unused variables in the example):
```
/Users/home/Desktop/app/src/utilities/filename.js
  1:7   error  'obj' is defined but never used   no-unused-vars
  2:25  error  'arg2' is defined but never used  no-unused-vars

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. This issue was originally reported in https://github.com/eslint/eslint/issues/7574 and was fixed in ESLint v3.10.0, which came out last year. You can get the fix by upgrading to a more recent version of ESLint.

Closing this issue as a duplicate.

**Comment by gitasong** [comment]
Ah, will do; thanks! I guess I didn't go back far enough with my search. :)
